src/app/auth/auth.component.ts

```diff
@@ -63,45 +65,93 @@ export class AuthComponent implements OnInit, OnDestroy {
           this.goToSessionSelect();
           return;
         }
-        this.githubService.storeOAuthAccessToken(token);
         this.authService.storeOAuthAccessToken(token);
-        this.userService.getAuthenticatedUser().subscribe((user: GithubUser) => {
-          auth.setLoginStatusWithGithub(true);
-          this.currentUserName = user.login;
-          if (this.isUserAuthenticating() || this.isAwaitingOAuthUserConfirm()) {
-            this.auth.changeAuthState(AuthState.ConfirmOAuthUser);
-          } else {
-            this.completeLoginProcess(this.currentUserName);
-          }
-        });
       });
     });
   }
 
   ngOnInit() {
-    this.checkAppIsOutdated();
-    this.authStateSubscription = this.auth.currentAuthState.subscribe((state) => {
-      this.authState = state;
-    });
-    this.loginForm = this.formBuilder.group({
-      username: ['', Validators.required],
-      password: ['', Validators.required],
-    });
-    this.profileForm = this.formBuilder.group({
-      session: ['', Validators.required],
-    });
+    this.isReady = false;
+    const oauthCode = this.activatedRoute.snapshot.queryParamMap.get('code');
+
+    if (this.authService.isAuthenticated()) {
+      this.router.navigate([this.phaseService.currentPhase]);
+      return;
+    }
+
+    if (oauthCode) { // In the web's oauth window
+      window.opener.postMessage({ oauthCode }, AppConfig.origin);
+      this.listenForCloseOAuthWindowMessage();
+    } else { // In the main app window
+      this.checkAppIsOutdated();
+      this.accessTokenSubscription = this.authService.accessToken.pipe(
```

Comment:
I suggest abstracting out the AuthComponent's initialisation logic into independent methods, so that `ngOnInit` is more readable.

We can refactor `ngOnInit` as follows:
```
  ngOnInit() {
    this.isReady = false;
    const oauthCode = this.activatedRoute.snapshot.queryParamMap.get('code');

    if (this.authService.isAuthenticated()) {
      this.router.navigate([this.phaseService.currentPhase]);
      return;
    }

    if (oauthCode) { // In the web's oauth window
      window.opener.postMessage({ oauthCode }, AppConfig.origin);
      this.listenForCloseOAuthWindowMessage();
    } else { // In the main app window
      this.checkAppIsOutdated();
      this.initAccessTokenSubscription();
      this.initAuthStateSubscription();
      this.initProfileForm();
    }
  }
```

The details of the initialisation will be in `initAccessTokenSubscription`, `initAuthStateSubscription`, `initProfileForm`
```